domain/model/service/modelservice.go

```diff
@@ -162,25 +178,44 @@ func (s *ModelService) CreateModel(
 	ctx context.Context,
 	controllerUUID uuid.UUID,
 ) error {
-	return errors.Capture(s.CreateModelForVersion(ctx, controllerUUID, agentVersionSelector()))
+	if err := createModelForVersion(
```

Comment:
This func name should include agent prefixed to version. Otherwise version has no meaning and adds confusion to the language we have around model.